### PR TITLE
fix: add Max Height for Select Component

### LINKED ISSUE #377: [Bug] Select various bugs
* [x ] I have searched the [issues](https://github.com/uber-web/baseui/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior
1. `multiple` prop should default to false
2. `rows` prop should indicate number of options shown in the dropdown list
3. each option needs an `id` in order for the selected styling to be applied appropriately
4. Keyboard shortcuts expose which `Option` override is focused so that customer styling can be added. (This may be a feature enhancement rather than a bug).

## Current Behavior
1. When the `multiple` prop is specified and when it is set (`multiple={false}`) I am still able to select several options
2. While using a custom `Option` component, the number of rows does not correlate to the number or custom options. For example, when `rows={3}` my dropdown only showed half of one option. It seems like this is configured for use with a very specific row height. 
3. When using a list of items, every time I selected one every single item in the dropdown would update with a checkmark. It was not until I looked at the source code for the dropdown until I realized that it determined this styling by checking for `selected.id === item.id`. It was unclear from the docs that each item needs an `id` to behave appropriately. Ideally this should be stated in the docs. 
4. When a custom option is used, the keyboard arrows will accurately select an item. However, the style of the focused item is overridden by the custom option. There is no prop that indicates which index is focussed from keyboard shortcuts, so I cannot apply styles to that option. 

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v2.2.1  |
| React   |    16.5.2     |
| browser |    Chrome     |
| node.js |         |
| etc.    |         |

## Steps to Reproduce

Link:

1.

2.

3.

4.

### PR DESCRIPTION
**Pull request by blackswanny**
Fixes #377 only bug 2

Expected Behavior
rows prop should indicate number of options shown in the dropdown list

Current Behavior

While using a custom Option component, the number of rows does not correlate to the number of custom options. For example, when rows={3} my dropdown only showed half of one option. It seems like this is configured for use with a very specific row height

### CODE REVIEW

File: rfcs/select-component.md [hunk 1]
```diff
@@ -153,6 +153,8 @@ export default () => {
   Whether the control should be in an error state. Default is `false`
 * `multiple: ?boolean`:
   Sets if multiple choices are allowed in Select component. Default is `false`
+* `maxHeight: ?string`:
```
Comment (nadiia) [inline rfcs/select-component.md]: We should also delete `rows` (line 146) from the API doc here

File: src/select/dropdown.js [hunk 2]
```diff
@@ -65,6 +66,7 @@ export default function SelectDropDown(props: DropDownPropsT) {
       overrides={{
         List: {
           component: DropDown,
+          style: {maxHeight},
           props: {
             $rows: rows,
```
Comment (nadiia) [inline src/select/dropdown.js]: Is this one still needed?

File: src/select/select.js [hunk 3]
```diff
@@ -334,11 +336,17 @@ class Select extends React.Component<PropsT, StatelessStateT> {
 
   getDropDown() {
     const {overrides, type, rows, selectedOptions} = this.props;
+    let maxHeight = this.props.maxHeight;
+    if (__BROWSER__ && parseInt(maxHeight) > window.innerHeight) {
```
Comment (nadiia) [inline src/select/select.js]: Should we have some check to make sure we're dealing here with `maxHeight` pixels value?

File: src/select/select.js [hunk 4]
```diff
@@ -334,11 +336,17 @@ class Select extends React.Component<PropsT, StatelessStateT> {
...
+      // only for pixel-sized maxHeight
+      maxHeight = '80vh';
+    }
     const options = this.getOptions();
     const {isDropDownOpen, optionsLoaded} = this.state;
     const dropDownProps = {
       rows,
```
Comment (nadiia) [inline src/select/select.js]: We should remove `rows` is it's now used.

File: rfcs/select-component.md (hunk 1 above)
Comment (nadiia) [reply rfcs/select-component.md]: Since this property is exposed on the select, how about we name it something like `maxMenuHeight` or `maxDropdownHeight` so there is no confusion that it can be related to the select's height?

Comment (gergelyke) [reply rfcs/select-component.md]: I'd go with `maxMenuHeight` as https://react-select.com/props

File: src/select/dropdown.js (hunk 2 above)
Comment (blackswanny) [reply src/select/dropdown.js]: removed

File: src/select/select.js (hunk 4 above)
Comment (blackswanny) [reply src/select/select.js]: removed

**Comment by gergelyke** [comment]
are we looking at a breaking change with this one?

File: src/select/select.js (hunk 3 above)
Comment (blackswanny) [reply src/select/select.js]: added

File: rfcs/select-component.md (hunk 1 above)
Comment (gergelyke) [reply rfcs/select-component.md]: makes sense

**Comment by blackswanny** [comment]
@gergelyke added migration guide

File: docs/migrations/00397.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#### In `Select` component use `maxDropdownHeight` instead of `rows` to setup max height of Select DropDown
+
+##### `maxDropdownHeight` accepts any height style as a string. For `px` value it can prevent Dropdown from being hidden and always set to no more 90% of screen
```
Comment (gergelyke) [inline docs/migrations/00397.md]: I am not sure if I understand this part - can you rephrase? also may worth mentioning the default values?

Comment (blackswanny) [reply docs/migrations/00397.md]: rephrased

File: src/select/__tests__/styled-components.test.js [hunk 6]
```diff
@@ -61,23 +61,21 @@ describe('Select styled components', () => {
...
+      (prop, value) => {
+        const props = {};
+        props[prop] = value;
+        const component = shallow(
+          <StyledDropDown {...props}>
+            <div />
+          </StyledDropDown>,
+        );
+        expect(component.instance().getStyles()).toMatchSnapshot(
+          'StyledDropDown has correct styles when ' + prop + ' is ' + value,
```
Comment (nadiia) [inline src/select/__tests__/styled-components.test.js]: Use string template instead concatenation.

Comment (blackswanny) [reply src/select/__tests__/styled-components.test.js]: ok
